From 06c287c760d8744ca007c1304aeae82d0a531b12 Mon Sep 17 00:00:00 2001
From: Jiang Lu <lu.jiang@windriver.com>
Date: Wed, 18 May 2011 17:49:47 +0800
Subject: [PATCH] ARM:CNS3xxx:Add oprofile support.

The CNS3xxx processor uses int 48 as all arm core pmu interrupt.
The CNS3420 uses kernel's op_model_arm11_core model as oprofile
model.

Signed-off-by: Jiang Lu <lu.jiang@windriver.com>
---
 arch/arm/kernel/pmu.c                     |    2 ++
 arch/arm/mach-cns3xxx/include/mach/irqs.h |    2 ++
 2 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/arch/arm/kernel/pmu.c b/arch/arm/kernel/pmu.c
index a900715..c74f867 100644
--- a/arch/arm/kernel/pmu.c
+++ b/arch/arm/kernel/pmu.c
@@ -46,6 +46,8 @@ static const int irqs[] = {
 	IRQ_PMU,
 #elif defined(CONFIG_ARCH_MXC)
 	IRQ_PMU_MXC,
+#elif defined(CONFIG_ARCH_CNS3XXX)
+	IRQ_PMU_CNS3XXX,
 #endif
 };
 
diff --git a/arch/arm/mach-cns3xxx/include/mach/irqs.h b/arch/arm/mach-cns3xxx/include/mach/irqs.h
index 2ab96f8..6dc6ead 100644
--- a/arch/arm/mach-cns3xxx/include/mach/irqs.h
+++ b/arch/arm/mach-cns3xxx/include/mach/irqs.h
@@ -15,6 +15,8 @@
 #define IRQ_LOCALWDOG		30
 #define IRQ_TC11MP_GIC_START	32
 
+#define IRQ_PMU_CNS3XXX 48
+
 #include <mach/cns3xxx.h>
 
 #ifndef NR_IRQS
-- 
1.7.0.4

